core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -263,9 +263,31 @@ private[sql] final case class ColumnTableScan(
     }
   }
 
+  def splitMoveNextMethods(ctx: CodegenContext, blocks: ArrayBuffer[String],
+      arg: String): String = {
+    if (blocks.length == 1) {
+      // inline execution if only one block
+      blocks.head
+    } else {
```

Comment:
This change is to force inlining irrespective of JIT decision. But in our case, this case won't arise as we will be calling splitMethod only in case of the wide schema. Hence that piece code is redundant.  Will remove.